luacompiler/src/lib/bytecode/mod.rs

```diff
@@ -1,36 +1,160 @@
 pub mod instructions;
 
+use self::instructions::{format_instr, HLInstr};
 use bincode::{deserialize, serialize};
-use irgen::constants_map::ConstantsMap;
+use irgen::{constants_map::ConstantsMap, register_map::Lifetime};
 use std::{
     fmt,
     fs::File,
     io::{self, Write},
     vec::Vec,
 };
 
+/// Represents a function in Lua.
+#[derive(Serialize, Deserialize)]
```

Comment:
I know I should have spotted these earlier, but why is this Serialize/Deserialize?

Reply:
The bytecode stores a vector of `Function`s. The compiler serializes the bytecode to a file, which means that `Function`s must be serializable as well.

Reply:
Is the representation that's best for the compiler the representation that's best for the VM?

Reply:
In my implementation I am using all the information that is provided in this structure. I will look into a more compact way of encoding this for the VM, but until I have a basic/naive implementation I can't really decide what to exclude.

Reply:
That's not quite my point. Compilers and VMs have different requirements of these sorts of things: what's easy to work with for a compiler might not be easy to work with for the VM (and vice versa). I'm trying to work out if we're optimising for one use case or the other (or maybe making both hard!). Things like the `id` field do not feel natural for the compiler, for example.